.github/workflows/lint.yaml

```diff
@@ -1,12 +1,17 @@
 name: Lint
 
 on: pull_request
-permissions: read-all
+
+permissions:
+  actions: read
+  checks: write
```

Comment:
Just so I'm aware -- what do we need `write` on checks for?

Reply:
It was initially mentioned in [this thread](https://masterpoint.slack.com/archives/C035S6TE7PC/p1697462944163189) that this action need write permissions to post annotations.
I've just reproduced in for this repo https://github.com/masterpointio/taskit/actions/runs/6669078751/job/18126063384#step:3:338